airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java

```diff
@@ -295,6 +295,47 @@ protected void syncShouldHandlePurgedLogsGracefully() throws Exception {
         "Expected 46 records to be replicated in the second sync.");
   }
 
+  /**
+   * This test verifies that multiple states are sent during the CDC process based on number of records.
+   * We can ensure that more than one `STATE` type of message is sent, but we are not able to assert
+   * the exact number of messages sent as depends on Debezium.
+   *
+   * @throws Exception Exception happening in the test.
+   */
+  @Test
+  protected void verifyCheckpointStatesByRecords() throws Exception {
+    // We require a huge amount of records, otherwise Debezium will notify directly the last offset.
```

Comment:
IIRC there is a config option to control the number of state messages? so that it can be set to something smaller : example : https://github.com/airbytehq/airbyte/blob/a9dacb92e7a81948b537c53b1c061086cf75ee8e/airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java#L106-L106

Can we use the same config to override the default state emission frequency for cdc incremental runs?

Reply:
If you set that in the config, it will be respected : 
https://github.com/airbytehq/airbyte/blob/a9dacb92e7a81948b537c53b1c061086cf75ee8e/airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/AirbyteDebeziumHandler.java#L117-L117

Reply:
By default for the tests we are already setting the `sync_checkpoint_records` to 1
`.put("sync_checkpoint_records", 1)` ([here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java#L103))

Reply:
We already overriding it to 1 like Subodh said.

Reply:
I see. I guess my question is why do we need a large number of records to create to test this? Is this because the offset (bin log file/pos) doesn't advance unless we are inserting such a large amount of records?

Reply:
That is a good question. I don't think we need that many, but 20k records isn't a lot of records anyway. We can reduce this if we need to. @subodh1810 thoughts?

Reply:
We need so many records because the offset file might not always be available to read (being edited by Debezium in parallel)